pymc_experimental/model/transforms/autoreparam.py

```diff
@@ -174,15 +176,18 @@ def vip_reparam_node(
 ) -> Tuple[ModelDeterministic, ModelNamed]:
     if not isinstance(node.op, RandomVariable | SymbolicRandomVariable):
         raise TypeError("Op should be RandomVariable type")
-    rv = node.default_output()
-    try:
-        [rv_shape] = constant_fold([rv.shape])
-    except NotConstantValueError:
-        raise ValueError("Size should be static for autoreparametrization.")
+    _, size, *_ = node.inputs
+    eval_size = size.eval()
+    if eval_size is not None:
+        rv_shape = tuple(eval_size)
+    else:
+        rv_shape = ()
+    lam_name = f"{name}::lam_logit__"
+    _log.debug(f"Creating {lam_name} with shape of {rv_shape}")
```

Comment:
Do we need the log or is it just temporary for debugging?

Reply:
this log level is not printed by default, I decided to add that, since the shape can change some day, and this log message could be helpful.

Reply:
The debug logging is a very nice thing, you rarely need that, but when you do..., we have a very bad handling of this functionality so far, it can be improved one day